app/models/katello/repository.rb

```diff
@@ -742,6 +743,19 @@ def self.linked_repositories
       to_return
     end
 
+    def self.search_by_redhat(_key, operator, value)
+      value = value == 'true'
+      value = !value if operator == '<>'
+
+      product_ids = Katello::Product.redhat.pluck(:id)
```

Comment:
can you use select here so that we aren't issuing a second query?

Reply:
Not sure I understand @jlsherrill. Are you suggesting to use
```ruby
product_ids = Katello::Product.redhat.select(:id).map { |p| p.id }
```
instead? They generate the same query.